hermes.py

```diff
@@ -212,7 +212,13 @@ def extract_bucket_key_from_s3_url(url):
 def decrypt_and_write_to_file(encrypted_filename, decrypted_output_filename, secret_key_file):
     decrypted_dict = None
     with open(encrypted_filename, "r") as f:
-        decrypted_dict = asym_crypto_yaml.load(f, secret_key_file)
+        for secret_key in secret_key_file:
+            try:
+                decrypted_dict = asym_crypto_yaml.load(f, secret_key)
+                break
+            except Exception as err:
+                logging.error('ERROR decrypting File %s: %s\n' % (str(f.name), str(err)))
```

Comment:
Can you log an info or a warning here instead of an error?

It is not an error to fail to decrypt with a given key when we are rotating keys.


Can you also add an error if the decryption fails with ALL keys specified for the file?

That is an error.  We have alerts in splunk that notify whenever the 'ERROR' text is logged by hermes.



Before doing the decryption you'll need to do the string comma split yourself